docs/shrc_diag_util.md

```diff
@@ -22,104 +22,124 @@
 
-Displays the contents of the cache in chronological order. You can use this output to see the history of updates that were made to the cache. For layered caches, some information is shown for the top layer cache only, and some is shown for all layers combined. To see information for the top layer cache only, use [`printTopLayerStats=all`](#printtoplayerstats).
+For a persistent cache, initialization problems can also occur if there is insufficient disk space available or if you are attempting to locate the cache on a remote networked file system, which is not supported.
 
-Each entry in the output starts with a VM ID, so you can see which VM wrote the associated data. Here are example entries for various types of cache data, with explanations:
+For a non-persistent cache, a shared memory area is required. Check that your system is configured with sufficient resources as described in [Setting shared memory values](configuring.md#setting-shared-memory-values). If initialization problems persist, try deleting the cache by using the `-Xshareclasses:destroyAll` cache utility, which removes all known memory areas and semaphores. Also remove the `javasharedresources` directory. These action force the VM to re-create the cache.
 
-### Class paths
+## Problems finding or storing classes
 
-: This example shows one class path with 4 entries:
+Classes are typically not found if the class is stale, as described in [Class data sharing operations](shrc.md#class-data-sharing-operations). Stale classes are redeemed if the same class is subsequently fetched by the class loader from another VM and checked against the stale class in the cache.
 
-        1: 0x2234FA6C CLASSPATH
-                /myVM/Apps/application1.jar
-                /myVM/Apps/application2.jar
-                /myVM/Apps/application3.jar
-                /myVM/Apps/application4.jar
+If you are attempting to share modified bytecode, you must use a modification context, as described in [Sharing modified bytecode](shrc.md#sharing-modified-bytecode). Otherwise, classes are stored in a private area that is not accessible to other VMs.
 
-    - `1` : the ID of the VM that wrote this data.
-    - `0x2234FA6C` : the address where this data is stored.
-    - `CLASSPATH` : the type of data that was written.
+If you are using a custom class loader, class path entries in the `SharedClassURLClasspathHelper` must be *confirmed* before classes can be found for these entries. More information about confirmed entries is available in the `SharedClassHelper` [API documentation](api-overview.md#monitoring-and-management).
```

Comment:
I found it in the `SharedClassURLClasspathHelper` interface, which extends `SharedClassHelper`. I can't link directly to is from this topic and shouldn't, because I'd have to choose between Java 8 or 11 ... and in future we'd have to update etc. When you said you couldn't find it, hopefully you just meant the mention of *confirmed* classes and not the `com.ibm.oti.shared` package from the link provided? 

Anyway, I updated the text to be more specific:

*If you are using a custom class loader, class path entries in the `SharedClassURLClasspathHelper` must be *confirmed* before classes can be found for these entries. More information about confirmed entries is available in the `SharedClassURLClasspathHelper` interface in the `com.ibm.oti.shared` API documentation*

Reply:
> you just meant the mention of confirmed classes and not the com.ibm.oti.shared package from the link provided?

Yes, I meant the mention of confirmed entries.